### ISSUE TITLE
fairmq-shmmonitor -c doesn't clean up

### ISSUE THREAD

**Issue by teo** [FairRootGroup/FairMQ#330]
This was reproduced on O²/FLP cluster at LHC Point 2, FLP 028.
No FairMQ devices currently running, multiple FairMQ-produced files in `/dev/shm`.
Ran `fairmq-shmmonitor -c` to perform an immediate cleanup, no success. See screenshot.

CC7, O²/FLP Suite v0.15.0, FairMQ v1.4.30

<img width="976" alt="Screenshot_20210218_170033" src="https://user-images.githubusercontent.com/126364/108384998-823a5600-720b-11eb-866d-19b11f6c31bb.png">

**Comment by rbx** [comment]
You need to pass either session id via `-s` or the shmid via `--shmid` (it's the string you see in the middle (`4cc14c93`)). Without these it will try to cleanup the session id `default` (that translates to the shmid you see in the log).

**Comment by teo** [comment]
Then perhaps we can turn this into a feature request: can you please give me a variation of `fairmq-shmmonitor -c` that will clean up **everything** on that machine that isn't currently used by any FairMQ device?

**Comment by rbx** [comment]
1) The cleanup tool only knows what is "not used" if it has the session details. And even these can be wrong under certain circumstances (if something crashed without decrementing its use count). So the "not used" info is not even considered by the `-c` command.
2) There is no list of all current sessions, by design. Sessions are independent and are the top level abstraction - there is nothing that knows all sessions on our side. You need to keep track of your session(s).
3) shmem files could be used by different users, or even by the same user, but belong to many different sessions.
4) If you are really sure it is safe to delete all, you could just run `rm -rf /dev/shm/*` or `rm -rf /dev/shm/fmq*; rm -rf /dev/shm/sem.fmq*`. But I would avoid adding these to the tool, because that can easily go wrong and cleanup something one doesn't want.

**Comment by ironMann** [comment]
Here is the same issue on EPNs. From what I am withnessing is that creation of new UnmanagedRegions fail. We have leftover `fairmq-shmmonitor` processes running. Applications that use UnmanagedRegion functionality (usually more than half of RAM) are  prevented from starting again. 

```
# ls /dev/shm/ -ltra

-rw-r--r--  1 epn  epn            32 Feb 19 21:01 sem.fmq_baf67292_ms
-rw-r--r--  1 epn  epn            32 Feb 19 21:05 sem.fmq_baf67292_mtx
-rw-r--r--  1 epn  epn       6553600 Feb 19 21:05 fmq_baf67292_mng
-rw-r--r--  1 epn  epn    2147483648 Feb 19 21:05 fmq_baf67292_m_0
-rw-r--r--  1 epn  epn           104 Feb 19 21:05 fmq_baf67292_cv
-rw-r--r--  1 epn  epn     536870912 Feb 19 21:05 fmq_baf67292_rg_1
-rw-r--r--  1 epn  epn  206158430208 Feb 19 21:05 fmq_baf67292_rg_2
-rw-r--r--  1 epn  epn       6316240 Feb 19 21:05 fmq_baf67292_rgq_2
-rw-r--r--  1 epn  epn       6316240 Feb 19 21:06 fmq_baf67292_rgq_1
-rw-r--r--  1 epn  epn        280208 Feb 20 10:54 fmq_baf67292_cq

# ps aux | grep fairmq

epn        14240  0.0  0.0 107168  2472 ?        Ssl  Feb19   0:02 /opt/alisw/el8/FairMQ/v1.4.30-5/bin/fairmq-shmmonitor -x --shmid baf67292 -d -t 2000
epn        49968  0.0  0.0 107168  3152 ?        Ssl  Feb19   0:03 /opt/alisw/el8/FairMQ/v1.4.30-5/bin/fairmq-shmmonitor -x --shmid baf67292 -d -t 2000

# lsof | grep fmq

COMMAND       PID    TID TASKCMD             USER   FD      TYPE             DEVICE SIZE/OFF       NODE NAME
fairmq-sh   14240                             epn  DEL       REG               0,22               80483 /dev/shm/fmq_baf67292_cq
fairmq-sh   14240  14241 fairmq-sh            epn  DEL       REG               0,22               80483 /dev/shm/fmq_baf67292_cq
fairmq-sh   14240  14242 fairmq-sh            epn  DEL       REG               0,22               80483 /dev/shm/fmq_baf67292_cq
fairmq-sh   49968                             epn  mem       REG               0,22   280208     119502 /dev/shm/fmq_baf67292_cq
fairmq-sh   49968  49969 fairmq-sh            epn  mem       REG               0,22   280208     119502 /dev/shm/fmq_baf67292_cq
fairmq-sh   49968  49970 fairmq-sh            epn  mem       REG               0,22   280208     119502 /dev/shm/fmq_baf67292_cq


# fairmq-shmmonitor -i

Starting shared memory monitor for session: "default" (shmId: baf67292)...
fairmq-shmmonitor for shared memory id baf67292 already started or not properly exited. Try `fairmq-shmmonitor --cleanup --shmid baf67292`
```

**Comment by rbx** [comment]
Leftover shmmonitor has been reported by Andrey (#331) and is fixed in v1.4.31. Please upgrade and retry.
That should fix the leftover processes at least.

If UnmanagedRegion creation still fails, please open a new issue, with more details.

**Comment by rbx** [comment]
I am closing this for the reasons stated above.
